pkg/datapath/linux/bandwidth/bandwidth.go

```diff
@@ -39,44 +62,123 @@ func GetBytesPerSec(bandwidth string) (uint64, error) {
 	return uint64(res.Value() / 8), err
 }
 
-func ProbeBandwidthManager() {
-	if option.Config.DryMode {
-		return
-	}
-
+// probe checks the various system requirements of the bandwidth manager and disables it if they are
+// not met.
+func (m *manager) probe() error {
 	// We at least need 5.1 kernel for native TCP EDT integration
 	// and writable queue_mapping that we use. Below helper is
 	// available for 5.1 kernels and onwards.
 	kernelGood := probes.HaveProgramHelper(ebpf.SchedCLS, asm.FnSkbEcnSetCe) == nil
-	option.Config.ResetQueueMapping = kernelGood
-	if !option.Config.EnableBandwidthManager {
-		return
+	m.resetQueues = kernelGood
+	if !m.params.Config.EnableBandwidthManager {
+		return nil
 	}
 	if _, err := sysctl.Read("net.core.default_qdisc"); err != nil {
-		log.WithError(err).Warn("BPF bandwidth manager could not read procfs. Disabling the feature.")
-		option.Config.EnableBandwidthManager = false
-		return
+		m.params.Log.WithError(err).Warn("BPF bandwidth manager could not read procfs. Disabling the feature.")
+		return nil
 	}
 	if !kernelGood {
-		log.Warn("BPF bandwidth manager needs kernel 5.1 or newer. Disabling the feature.")
-		option.Config.EnableBandwidthManager = false
-		return
+		m.params.Log.Warn("BPF bandwidth manager needs kernel 5.1 or newer. Disabling the feature.")
+		return nil
 	}
-	if option.Config.EnableBBR {
+	if m.params.Config.EnableBBR {
 		// We at least need 5.18 kernel for Pod-based BBR TCP congestion
 		// control since earlier kernels just clear the skb->tstamp upon
 		// netns traversal. See also:
 		//
 		// - https://lpc.events/event/11/contributions/953/
 		// - https://lore.kernel.org/bpf/20220302195519.3479274-1-kafai@fb.com/
 		if probes.HaveProgramHelper(ebpf.SchedCLS, asm.FnSkbSetTstamp) != nil {
-			log.Fatalf("Cannot enable --%s, needs kernel 5.18 or newer.",
-				option.EnableBBR)
+			return fmt.Errorf("cannot enable --%s, needs kernel 5.18 or newer", EnableBBR)
+		}
+	}
+
+	// Going via host stack will orphan skb->sk, so we do need BPF host
+	// routing for it to work properly.
+	if m.params.Config.EnableBBR && m.params.DaemonConfig.EnableHostLegacyRouting {
+		return fmt.Errorf("BPF bandwidth manager's BBR setup requires BPF host routing.")
```

Comment:
Bumping, I would like to understand that too. It seems a bit sketchy.

Reply:
> Why do we error out in this case instead of disabling bw management feature
This is a case of conflicting configuration both `EnableBBR` and `EnableHostLegacyRouting` are flags. And when we encounter incompatible configuration its standard practice in Cilium to crash the agent to notify the user of their configuration. (take `initKubeProxyReplacementOptions` for example)

>  similar to other probe cases?
Perhaps this is an odd corner case where these values are both flags and dynamically set. That is a less then ideal design decision. But since its both, I feel like we should assume the user specified this combination of configuration on purpose.

> I would like to understand that too
I don't know why this incompatibility exists at the moment, if that

Reply:
ok, I was more thinking about case `kernelGood` where we log warning and return nil, but for  `EnableBBR` which requires newer kernel, we return error - this seemed more inconsistent.

Reply:
This behavior is inconsistent in the whole agent. The code as is, is just moved but we have not changed the semantics, all of the decisions on when to crash, fallback, and report were already there. While I do agree that this isn't ideal, I also don't see a quick fix, it really depends on who you talk to. Some people want to crash other never, so we first need to figure out a policy and then apply that consistently. But I frankly don't think that should be part of this PR. So with your permission I would like to make an issue to discuss this further and then resolve the conversation here.

Reply:
https://github.com/cilium/cilium/issues/29169